src/pages/iou/MoneyRequestModal.js

```diff
@@ -113,11 +113,18 @@ const MoneyRequestModal = (props) => {
 
     useEffect(() => {
         PersonalDetails.openMoneyRequestModalPage();
-        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
         IOU.setMoneyRequestDescription('');
-        // eslint-disable-next-line react-hooks/exhaustive-deps -- props.currentUserPersonalDetails will always exist from Onyx and we don't want this effect to run again
     }, []);
 
+    // We update selected currency when PersonalDetails.openMoneyRequestModalPage finishes
+    // props.currentUserPersonalDetails might be stale data or might not exist if user is signing in
+    useEffect(() => {
+        if (_.isUndefined(props.currentUserPersonalDetails.localCurrencyCode)) {
+            return;
+        }
+        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
+    }, [props.currentUserPersonalDetails.localCurrencyCode]);
+
```

Comment:
@jczekalski 
`PersonalDetails.openMoneyRequestModalPage();` is called which is responsible for retrieving  `localCurrencyCode` for a user. But without waiting for its response the local state `onyx.iou` is updated with currently available data. Which is not available if signing in with slow network. That's why I had to make sure `iou` is updated when response data is available.

Any thoughts?

Reply:
I think this change makes sense and should solve the stale data issue. Code looks good in general. Could you prep the PR for review (complete the checklist, add videos for every platform etc.)?